stl/inc/xstring

```diff
@@ -4269,7 +4269,10 @@ template <class _Elem, class _Traits, class _Alloc>
 _NODISCARD basic_string<_Elem, _Traits, _Alloc> operator+(
     const basic_string<_Elem, _Traits, _Alloc>& _Left, const basic_string<_Elem, _Traits, _Alloc>& _Right) {
     // return string + string
-    basic_string<_Elem, _Traits, _Alloc> _Ans;
+    using _String_type = basic_string<_Elem, _Traits, _Alloc>;
+    // same as basic_string::_Alty_traits
+    using _Alty_traits = allocator_traits<typename allocator_traits<_Alloc>::template rebind_alloc<_Elem>>;
+    _String_type _Ans(_Alty_traits::select_on_container_copy_construction(_Left.get_allocator()));

```

Comment:
Allocator is copied at least 2 times there (maybe even 3 if no copy ellision occurs in `get_allocator`), firstly in `select_on_container_copy_construction` and then in `basic_string`'s constructor. There must be a way to replace this second copy with a move.

Reply:
Now there's only one copy thanks to your ctor overload